### PR TITLE
ArduPlane: add thr position trigger to ThrLandCtrl Q_OPTION

### PR DESCRIPTION
**Pull request by Hwurzburg**
…autolands

for use with systems like Swan Hwing when using throttle landing control option
also for use with gamepad style joysticks in MP control of quadplanes

SITL tested on Hwing

### CODE REVIEW

**Comment by Hwurzburg** [comment]
changed to thr position trigger to activate the ThrLandCntrl Q_OPTION during land descent phase

**Comment by Hwurzburg** [comment]
flight tested in RTL and re-tested in SITL in other modes

File: ArduPlane/config.h [hunk 1]
```diff
@@ -286,3 +286,9 @@
 #ifndef FS_EKF_THRESHOLD_DEFAULT
  # define FS_EKF_THRESHOLD_DEFAULT      0.8f    // EKF failsafe's default compass and velocity variance threshold above which the EKF failsafe will be triggered
 #endif
+
+/////////////////////////////////////////////////////////////////////////////
+//  Landing Throttle Control Trigger Threshold
+#ifndef THR_CTRL_LAND_THRESH
+ #define THR_CTRL_LAND_THRESH 0.7f
```
Comment (tridge) [inline ArduPlane/config.h:293]: please use 0.7 instead of 0.7f

File: ArduPlane/quadplane.h [hunk 2]
```diff
@@ -122,6 +122,9 @@ class QuadPlane
 
     // return true if the user has set ENABLE
     bool enabled(void) const { return enable != 0; }
+    
+    // is throttle controlled landing descent active?
+    mutable bool thr_ctrl_land;
```
Comment (tridge) [inline ArduPlane/quadplane.h:127]: instead of using mutable, better to just remove the const off QuadPlane::landing_descent_rate_cms()

**Review by tridge — CHANGES_REQUESTED** [review]

File: ArduPlane/quadplane.h (hunk 2 above)
Comment (Hwurzburg) [reply ArduPlane/quadplane.h]: done

File: ArduPlane/config.h (hunk 1 above)
Comment (Hwurzburg) [reply ArduPlane/config.h]: done

**Review by tridge — APPROVED** [review]

**Pull request merged**